.github/workflows/aks_deploy.yml

```diff
@@ -66,3 +66,43 @@ jobs:
         with:
           message: |
             Review app deployed to ${{ steps.deploy.outputs.url }}
+
+  test:
+    name: Run the test suite
+    uses: ./.github/workflows/rspec.yml
+
+  deploy_staging:
+    needs: [docker, test]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
```

Comment:
requires environment setting as you are using secrets in that format, so
```
environment:
    name: staging
```
Same for the sandbox and production deploys